projects/aca-shared/rules/src/app.rules.spec.ts

```diff
@@ -1003,6 +967,978 @@ describe('app.evaluators', () => {
+      expect(app.canCreateFolder(context)).toBeTrue();
+    });
+
+    it('should verify is user has create permission on current folder', () => {
+      context.appConfig = { get: () => true } as any;
+      context.navigation.url = '/personal-files/test';
+      context.navigation.currentFolder = { allowableOperations: ['create'] } as any;
+      spyOn(context.permissions, 'check');
+      app.canCreateFolder(context);
+      expect(context.permissions.check).toHaveBeenCalledWith(context.navigation.currentFolder, ['create']);
+    });
+  });
+
+  describe('canCreateLibrary', () => {
+    const context = createTestContext();
+
+    it('should return false when content service is disabled', () => {
+      context.appConfig = { get: () => false } as any;
+      expect(app.canCreateLibrary(context)).toBeFalse();
+    });
+
+    it('should return false when user is outside libraries', () => {
+      context.appConfig = { get: () => true } as any;
+      context.navigation.url = '/favorite/test';
+      expect(app.canCreateLibrary(context)).toBeFalse();
+    });
+
+    it('should return true when content service is enabled and user is in libraries', () => {
+      context.appConfig = { get: () => true } as any;
+      context.navigation.url = '/test/libraries';
+      expect(app.canCreateLibrary(context)).toBeTrue();
+    });
+  });
+
+  describe('canUpload', () => {
+    const context = createTestContext();
+
+    it('should return false when content service is disabled', () => {
+      context.appConfig = { get: () => false } as any;
+      expect(app.canUpload(context)).toBeFalse();
+    });
+
+    it('should return false when user is outside personal files or libraries', () => {
+      context.appConfig = { get: () => true } as any;
+      context.navigation.url = '/favorite/test';
+      expect(app.canUpload(context)).toBeFalse();
+    });
+
+    it('should return false when current folder does not exist', () => {
+      context.appConfig = { get: () => true } as any;
+      context.navigation.url = '/personal-files/test';
+      context.navigation.currentFolder = null;
+      expect(app.canUpload(context)).toBeFalse();
+    });
+
+    it('should return false when permission check fails', () => {
+      context.appConfig = { get: () => true } as any;
+      context.navigation.url = '/personal-files/test';
+      context.navigation.currentFolder = {} as any;
+      context.permissions = { check: () => false };
+      expect(app.canUpload(context)).toBeFalse();
+    });
+
+    it('should return true when permission requirements are met', () => {
+      context.appConfig = { get: () => true } as any;
+      context.navigation.url = '/personal-files/test';
+      context.navigation.currentFolder = {} as any;
+      context.permissions = { check: () => true };
+      expect(app.canUpload(context)).toBeTrue();
+    });
+
+    it('should verify is user has create permission on current folder', () => {
```

Comment:
Typo, should be 'should verify **if** user has create permissions...'

Reply:
Fixed